api/lib/domain/models/CampaignCompetenceCollectiveResult.js

```diff
@@ -0,0 +1,30 @@
+class CampaignCompetenceCollectiveResult {
+
+  constructor({
+    // attributes
+    campaignId,
+    competenceId,
+    competenceIndex,
+    competenceName,
+    totalSkillsCount,
+    averageValidatedSkills,
+  } = {}) {
+    // attributes
+    this.campaignId = campaignId;
+    this.competenceId = competenceId;
+    this.competenceIndex = competenceIndex;
+    this.competenceName = competenceName;
+    this.totalSkillsCount = totalSkillsCount;
+    this.averageValidatedSkills = averageValidatedSkills;
+  }
+
+  get id() {
+    return `${this.campaignId}_${this.competenceId}`;
+  }
+
+  get domainCode() {
+    return this.competenceIndex.charAt(0);
```

Comment:
`this.competenceIndex.split('.')[0]` semble plus résilient à l'ajout de nouvelles compétences aux index > 9.x. Je pense en particulier à l'ouverture de Pix à d'autres usages.

Reply:
On veut récupérer le code du Domaine de Compétence dont fait partie une compétence donnée.
La référence DIGCOMP compte pour l'instant 5 Domaines (référencés de 1 à 5).
Pour récupérer ce domainCode on fait un raccourci en passant par le competenceIndex.
C'est pas top, mais pour l'instant son utilisation est ciblée que pour le front Orga sur 2 hbs.

Reply:
Ce que proposait @dherault c'est de faire fonctionner ce code pour des `competenceIndex` tels que `10.1`. Avec `this.competenceIndex.charAt(0)` ce qu'on récupère sera `1`, ce qui n'est pas correct. Avec `this.competenceIndex.split('.')[0]` on récupère bien `10`.

Reply:
C'est corrigé, merci !